packages/components/src/custom-gradient-picker/index.tsx

```diff
@@ -68,7 +83,8 @@ const GradientTypePicker = ( { gradientAST, hasGradient, onChange } ) => {
 					? {}
 					: { orientation: HORIZONTAL_GRADIENT_ORIENTATION } ),
 				type: 'linear-gradient',
-			} )
+				// NTS: this isn't very elegant, but the `orientation` props between linear and radial gradients don't overlap (according to DefinitelyTyped) so TypeScript gets confused by this manual change.
+			} as gradientParser.LinearGradientNode )
```

Comment:
This doesn't feel very elegant, but the `orientation` props between linear and radial gradients don't overlap (according to DefinitelyTyped) so TypeScript gets confused by this manual change.

Reply:
> `value`, according to the `gradient-parser` types, should be a `string`, not a `number`

Shoot - I remember noticing that and then I must have lost track of it while testing different solutions. Good catch, thank you for highlighting it.

> Here's how we could apply a few tweaks and improve the correctness of the code, while making TS happy

Thanks for the diff, very helpful! A question for my own education... Is the `const { orientation, ...restGradientAST } = gradientAST` destructuring, a functional change or more for readability? It looks like the main functional change you've suggested is replacing `{}` with `undefined` if we don't have an `orientation`.

<details>
<summary>I wondered if this would work as well as your diff, or if there's a nuance I didn't see:</summary>

```diff

diff --git a/packages/components/src/custom-gradient-picker/constants.ts b/packages/components/src/custom-gradient-picker/constants.ts
index 9924ff2b3d..842aea9dcb 100644
--- a/packages/components/src/custom-gradient-picker/constants.ts
+++ b/packages/components/src/custom-gradient-picker/constants.ts
@@ -10,8 +10,8 @@ export const DEFAULT_LINEAR_GRADIENT_ANGLE = 180;
 
 export const HORIZONTAL_GRADIENT_ORIENTATION = {
 	type: 'angular',
-	value: 90,
-};
+	value: '90',
+} as const;
 
 export const GRADIENT_OPTIONS = [
 	{ value: 'linear-gradient', label: __( 'Linear' ) },
diff --git a/packages/components/src/custom-gradient-picker/index.tsx b/packages/components/src/custom-gradient-picker/index.tsx
index 8c9b9c7908..e19ac3d55f 100644
--- a/packages/components/src/custom-gradient-picker/index.tsx
+++ b/packages/components/src/custom-gradient-picker/index.tsx
@@ -79,11 +79,11 @@ const GradientTypePicker = ( {
 		onChange(
 			serializeGradient( {
 				...gradientAST,
-				...( gradientAST.orientation
-					? {}
-					: { orientation: HORIZONTAL_GRADIENT_ORIENTATION } ),
+				orientation: gradientAST.orientation
+					? undefined
+					: HORIZONTAL_GRADIENT_ORIENTATION,
 				type: 'linear-gradient',
-			} as gradientParser.LinearGradientNode )
+			} )
 		);
 	};
 
```
</details>